lib/model/graphParser.js

```diff
@@ -0,0 +1,89 @@
+var Graph = require('./graph');
+var Connection = require('./connection');
+var Node = require('./node');
+var Port = require('./port');
+var CardinalPortPosition = require('./cardinalPortPosition');
+var IOType = require('../enum/ioType');
+var ObjectUtils = require('../utilities/objects');
+
+var setIfNotExist = function(id, obj, map) {
+  if (Object.keys(map).indexOf(id) !== -1) {
+    console.error('Duplicate entity id: ' + id);
+    return false;
+  }
+  map[id] = obj;
+  return true;
+};
+
+var GraphParser = function() {
+  this._nodes = {};
```

Comment:
* The angle bracket syntax is still nominally correct, but containers are not yet implemented, so the syntax is not used anywhere yet.
* Since there are not yet any containers, any test data that we have should have unique IDs.
* Global, because we don't have containers yet.